### PR TITLE
fix(workflow): Avoid incidents query if none are specified

### PR DESCRIPTION
**Pull request by scttcper**
Was querying all incidents when none were specified

### CODE REVIEW

File: src/sentry/integrations/slack/unfurl/metric_alerts.py [hunk 1]
```diff
@@ -63,15 +63,17 @@ def unfurl_metric_alerts(
...
-        i.identifier: i
-        for i in Incident.objects.filter(
-            incident_filter_query,
-            # Filter by integration organization here as well to make sure that
-            # we have permission to access these incidents.
-            organization__in=all_integration_orgs,
-        )
-    }
+    incident_map = {}
+    if bool(incident_filter_query):
```
Comment (leeandher) [inline src/sentry/integrations/slack/unfurl/metric_alerts.py:67]: Just looking at the Line 37 above, where `incident_filter_query` is initialized, does that mean `bool( Q() )` will return `False`, cause if not then I think this conditional would always succeed, right?

**Review by leeandher — COMMENTED** [review]
Just a sanity check before approving 👍

File: src/sentry/integrations/slack/unfurl/metric_alerts.py (hunk 1 above)
Comment (leeandher) [reply src/sentry/integrations/slack/unfurl/metric_alerts.py:67]: Okay just realized I can check this in a Python console, you're totally right `bool(Q()) == False`

**Review by leeandher — APPROVED** [review]
looks good to me 🚀

**Review by wedamija — APPROVED** [review]

**Pull request merged**